### ISSUE TITLE
building serial unit test to then skip them 

### ISSUE THREAD

**Issue by felixschurk** [IPPL-framework/ippl#248]
When a serial build of ippl is performed over e.g.
```sh
./ippl-build-scripts/999-build-everything -t serial -i -k -f
```
around 596 unit tests are build, however executed are only a hand full 6 or so.

I think this behavior is quite confusing and also leads to a loss in information, as then detecting the few successful run unit tests over all the skipped ones gets hard to tell. Part of the output look like:
```md
591/596 Test #591: ParticleSendRecv.SendAndRecieve<Parameters<double,Kokkos::Serial,Rank<6u> >> ....................***Skipped   0.94 sec
        Start 595: ParameterListTest.Merge
592/596 Test #592: ParticleSendRecv.SendAndRecieve<Parameters<float,Kokkos::Serial,Rank<6u> >> .....................***Skipped   0.96 sec
        Start 596: ParameterListTest.Update
593/596 Test #593: ParameterListTest.Add ...........................................................................   Passed    0.93 sec
594/596 Test #594: ParameterListTest.UpdateSingle ..................................................................   Passed    0.84 sec
595/596 Test #595: ParameterListTest.Merge .........................................................................   Passed    0.73 sec
596/596 Test #596: ParameterListTest.Update ........................................................................   Passed    0.75 sec

100% tests passed, 0 tests failed out of 596

Total Test time (real) = 131.91 sec

The following tests did not run:
	  1 - BareFieldTest.DeepCopy<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  2 - BareFieldTest.Sum<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  3 - BareFieldTest.Min<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  4 - BareFieldTest.Max<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  5 - BareFieldTest.Prod<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  6 - BareFieldTest.ScalarMultiplication<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
	  7 - BareFieldTest.DotProduct<Parameters<double,Kokkos::Serial,Rank<1u> >> (Skipped)
```

As it is somehow possible to skip them, we might also directly not build the ones that can not be executed.

**Comment by Arc676** [comment]
Serial tests are skipped by default because I was worried about their reduced speed compared to parallel execution and thus the increased runtime of the unit tests. This turned out to be unfounded in the end, as even the serial execution tests run pretty quickly. If I remember correctly, we agreed that we could remove the test skipping entirely, but it hasn't been done yet.

**Comment by srikrrish** [comment]
The serial tests are not skipped now (the flag is changed to `false`) and this will be available from release 3.2.0 onwards

**Comment by Arc676** [comment]
In the interest of avoiding unnecessary clutter and redundant issues, I think this issue should stay open until the relevant code is actually removed. Changing the flag to `false` by default without changing anything else just means that there is now a lot of dead code that does nothing. I would not consider this a "solution" to the issue described here.
